convert-hf : print output file name when completed

danbev opened the pull request:
This commit adds the output file name to the log message when the conversion is completed.

The motivation for this change is that when `--outfile` option is not specified it migth not be obvious where the output file is written.

With this change the output of running the script will be something like the following:
```console
INFO:hf-to-gguf:Model successfully exported to models/gemma-2-9b-it.gguf.
```



- [x] I have read the [contributing guidelines](https://github.com/ggerganov/llama.cpp/blob/master/CONTRIBUTING.md)
- Self-reported review complexity:
  - [x] Low
  - [ ] Medium
  - [ ] High

danbev committed:
convert-hf : print output file name when completed

This commit adds the output file name to the log message when the
conversion is completed.

The motivation for this change is that when `--outfile` option is not
specified it migth not be obvious where the output file is written.

With this change the output of running the script will be something like
the following:
```console
INFO:hf-to-gguf:Model successfully exported to models/gemma-2-9b-it.gguf.
```

Signed-off-by: Daniel Bevenius <daniel.bevenius@gmail.com>

compilade commented on convert-hf-to-gguf.py:
```diff
@@ -3128,11 +3128,11 @@ def main() -> None:
             logger.info("Exporting model...")
             model_instance.write()
-            logger.info("Model successfully exported.")
+            logger.info(f"Model successfully exported to {fname_out}.")
```
This should use the name from the model, because `fname_out` otherwise could contain an unexpanded outtype template (like in its default value).

Also, the trailing dot can be confused as being part of the file name.
```suggestion
            logger.info(f"Model successfully exported to {model_instance.fname_out}")
```

The same comment also applies to the other changed line.

compilade reviewed (commented):
This is nice, although since #6942, the output file names are already printed before the progress bar.

On `master`, after printing the tensors, their types and their shape, there's this:
```
INFO:gguf.gguf_writer:Writing the following files:
INFO:gguf.gguf_writer:/srv/LLMstash/tmp/bf16-tests/tinymistral-bf16.gguf: n_tensors = 111, total_size = 496.1M
Writing: 100%|█████████████████████████████████████████████████████████████████████████████████| 496M/496M [00:05<00:00, 94.4Mbyte/s]
INFO:hf-to-gguf:Model successfully exported.
```

Your proposal would change this into

```
INFO:gguf.gguf_writer:Writing the following files:
INFO:gguf.gguf_writer:/srv/LLMstash/tmp/bf16-tests/tinymistral-bf16.gguf: n_tensors = 111, total_size = 496.1M
Writing: 100%|█████████████████████████████████████████████████████████████████████████████████| 496M/496M [00:05<00:00, 94.4Mbyte/s]
INFO:hf-to-gguf:Model successfully exported to /srv/LLMstash/tmp/bf16-tests/tinymistral-bf16.gguf.
```
Where the output file name is present twice.

And when using split options, the final name would not exist.

```
INFO:gguf.gguf_writer:Writing the following files:
INFO:gguf.gguf_writer:/srv/LLMstash/tmp/gguf-split-tests/tinymistral-100M/tinymistral-q8_0-00001-of-00003.gguf: n_tensors = 17, total_size = 99.5M
INFO:gguf.gguf_writer:/srv/LLMstash/tmp/gguf-split-tests/tinymistral-100M/tinymistral-q8_0-00002-of-00003.gguf: n_tensors = 58, total_size = 99.5M
INFO:gguf.gguf_writer:/srv/LLMstash/tmp/gguf-split-tests/tinymistral-100M/tinymistral-q8_0-00003-of-00003.gguf: n_tensors = 36, total_size = 64.7M
Shard (3/3): 100%|███████████████████████████████████████████████████████████████████████████| 64.7M/64.7M [00:00<00:00, 70.0Mbyte/s]
Writing: 100%|█████████████████████████████████████████████████████████████████████████████████| 264M/264M [00:07<00:00, 33.4Mbyte/s]
INFO:hf-to-gguf:Model successfully exported to /srv/LLMstash/tmp/gguf-split-tests/tinymistral-100M/tinymistral-q8_0.gguf.
$ ls /srv/LLMstash/tmp/gguf-split-tests/tinymistral-100M/
tinymistral-q8_0-00001-of-00003.gguf  tinymistral-q8_0-00002-of-00003.gguf  tinymistral-q8_0-00003-of-00003.gguf
```

But it might still be useful when not splitting.

danbev committed:
squash! convert-hf : print output file name when completed

Updates the output of to support printing the directory if the output is
split into multiple files. Also the output file name is now retrieved
from the model_instance object.

Signed-off-by: Daniel Bevenius <daniel.bevenius@gmail.com>

danbev commented:
> This is nice, although since #6942, the output file names are already printed before the progress bar.

Ah sorry, I see it now, for some reason I just missed it :(  

Now that I know where to look I'm happy with not having the output file name/path in the final output, but perhaps it might be nice to have it just the same in case others run into the same issue I did. I'll push a commit with a suggestion with the changes you suggested and also include printing the directory when the conversion performs a split. Thanks!

compilade commented on convert-hf-to-gguf.py:
```diff
@@ -3128,11 +3129,12 @@ def main() -> None:
             logger.info("Exporting model...")
             model_instance.write()
-            logger.info("Model successfully exported.")
+            out_path = os.path.dirname(model_instance.fname_out) + '/' if is_split else model_instance.fname_out
```
`model_instance.fname_out` is a `Path` (from `pathlib`), so instead of `os.path.dirname`, there's also the `parent` property:
```suggestion
            out_path = f"{model_instance.fname_out.parent}/" if is_split else model_instance.fname_out
```

danbev replied:
Updated now :+1:

compilade replied:
I think the trailing slash should be `{os.sep}`. I don't have a Windows development environment, so I'm not sure about this, but from jumping around its source, `pathlib` uses the separator of the host OS when printing with `str(some_path)` (like implicitly in `f"{some_path}"`), and so always using a slash (`/`) might look weird on that OS.

If that's not true, feel free to disregard this.

danbev replied:
Good point! I'll take a look what this looks like on Windows.

danbev replied:
This does not look correct on Windows (just using `/`) and I'll update it to use `os.sep` shortly. Thanks!

compilade reviewed (commented):
> Ah sorry, I see it now, for some reason I just missed it :(

> but perhaps it might be nice to have it just the same in case others run into the same issue I did.

I agree, this should reduce confusion.

Good idea with the output directory name when splitting.

danbev committed:
squash! convert-hf : print output file name when completed

Use parent attribute of Path object and string interpolation.

Signed-off-by: Daniel Bevenius <daniel.bevenius@gmail.com>

danbev committed:
squash! convert-hf : print output file name when completed

Use os.sep instead of hardcoding the path separator.

Signed-off-by: Daniel Bevenius <daniel.bevenius@gmail.com>

compilade reviewed (approved):
Looks good to me.
